game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/BattleSteps.java

```diff
@@ -0,0 +1,242 @@
+package games.strategy.triplea.delegate.battle.steps;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.GamePlayer;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.Properties;
+import games.strategy.triplea.attachments.TechAttachment;
+import games.strategy.triplea.attachments.UnitAttachment;
+import games.strategy.triplea.delegate.Matches;
+import games.strategy.triplea.delegate.battle.BattleStepStrings;
+import games.strategy.triplea.delegate.battle.MustFightBattle.ReturnFire;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import lombok.Builder;
+import lombok.NonNull;
+import org.triplea.java.collections.CollectionUtils;
+
+/** Get the steps that will occurr in the battle */
+@Builder
+public class BattleSteps implements BattleStepStrings {
```

Comment:
Would you mind describing the reasoning to implement BattleStepStrings?

Reply:
Sure.  Almost all of the constants in that class are used by `determineStepStrings`.  Since I moved that code into a new class, I thought I would just implement it just like MustFightBattle.  I'm not sure why it was implemented by MustFightBattle vs importing the constants directly and just wanted to follow the existing convention.